src/validators.js

```diff
@@ -1,20 +1,44 @@
 
 const assert = require('assert');
 
+
 function validateApiKey(value) {
-    const typeError = new TypeError('ImgixAPI.settings.apiKey must be passed a string');
+    const invalidApiKeyError = new TypeError('ImgixAPI.settings.apiKey must be passed a string');
     const legalKey = /[0-9a-f]{64}/;
     const legalKeyError = new TypeError(`${value} does not match a legal apiKey structure`);
 
-    assert(typeof value === 'string', typeError);
+    assert(typeof value === 'string', invalidApiKeyError);
     assert(legalKey.exec(value), legalKeyError);
 };
 
 function validateOpts(options) {
     validateApiKey(options.apiKey);
 };
 
+function validateBody(body) {
+    const invalidBodyError = new TypeError('The request body must a valid JSON object or a Buffer');
+
+    const isValid = body && (isJSON(body) || isBuffer(body));
+    assert(isValid, invalidBodyError);
+};
+
+function isBuffer(body) {
+    return Buffer.isBuffer(body);
+};
+
+function isJSON(body) {
+    try {
+        JSON.parse(body);
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
```

Comment:
Waaaait, I just realized we're requiring users to stringify their JSON before passing it to us. Let's not do that! Let's instead validate that the given value is an object and that it *can* be stringified, and then we'll stringify it for them.